bciers/apps/compliance/src/tests/components/compliance-summary/manage-obligation/apply-compliance-units/ApplyComplianceUnitsComponent.test.tsx

```diff
@@ -304,6 +304,122 @@ describe("ApplyComplianceUnitsComponent", () => {
     });
   });
 
+  it("handles invalid account ownership during submit", async () => {
+    // Mock valid account details first
+    (getBccrAccountDetails as ReturnType<typeof vi.fn>).mockResolvedValueOnce({
+      bccr_trading_name: MOCK_TRADING_NAME,
+    });
+
+    // Mock invalid ownership response (error) for the submit API call
+    (actionHandler as ReturnType<typeof vi.fn>).mockResolvedValueOnce({
+      error:
+        "The provided holding account does not own the compliance sub-account for this operation.",
+      bccr_units: [],
+    });
+
+    render(
+      <ApplyComplianceUnitsComponent
+        complianceReportVersionId={TEST_COMPLIANCE_REPORT_VERSION_ID}
+        reportingYear={2024}
+      />,
+    );
+
+    // Enter account ID
+    const accountInput = screen.getByLabelText("BCCR Holding Account ID:*");
+    fireEvent.change(accountInput, { target: { value: VALID_ACCOUNT_ID } });
+
+    await waitFor(() => {
+      expect(screen.getByText(MOCK_TRADING_NAME)).toBeVisible();
+    });
+
+    // Check confirmation checkbox
+    const checkbox = screen.getByRole("checkbox");
+    fireEvent.click(checkbox);
+
+    await waitFor(() => {
+      expect(checkbox).toBeChecked();
+    });
+
+    // Submit form
+    const submitButton = screen.getByRole("button", { name: "Submit" });
+    fireEvent.click(submitButton);
+
+    // Should show ownership validation error
+    await waitFor(() => {
+      expect(
+        screen.getByText(
+          "The provided holding account does not own the compliance sub-account for this operation.",
+        ),
+      ).toBeVisible();
+    });
+  });
+
+  it("prevents form progression when account ownership validation fails at initial step", async () => {
+    (getBccrAccountDetails as ReturnType<typeof vi.fn>).mockRejectedValueOnce(
+      new Error(
+        "The provided holding account does not own the compliance sub-account for this operation.",
+      ),
+    );
+
+    render(
+      <ApplyComplianceUnitsComponent
+        complianceReportVersionId={TEST_COMPLIANCE_REPORT_VERSION_ID}
+        reportingYear={2024}
+      />,
+    );
+
+    // Enter account ID
+    const accountInput = screen.getByLabelText("BCCR Holding Account ID:*");
+    fireEvent.change(accountInput, { target: { value: VALID_ACCOUNT_ID } });
+
+    // Should show error message from BccrHoldingAccountWidget
+    await waitFor(() => {
+      expect(
+        screen.getByText(
+          "The provided holding account does not own the compliance sub-account for this operation.",
+        ),
+      ).toBeVisible();
+    });
+
+    // Should not show confirmation checkbox or allow progression
+    expect(screen.queryByRole("checkbox")).not.toBeInTheDocument();
+    expect(screen.queryByText("BCCR Trading Name:")).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole("button", { name: "Submit" }),
+    ).not.toBeInTheDocument();
+  });
+
+  it("handles account ownership validation API errors gracefully", async () => {
```

Comment:
Also testing the same thing as `it("handles account validation errors", async () => {`